test/unit/EchoClientTest.php

```diff
@@ -57,18 +57,142 @@ function testEventWritten()
         $mockRequest = $this->getMock('\Guzzle\Http\Message\Request', array('send'), array('post',''));
         $mockRequest->expects($this->once())->method('send')->will($this->returnValue($response));
 
+        $expectedHeaders = array(
+            'Content-Type' => 'application/json',
+            'Authorization' => 'Bearer some-token'
+        );
+        $expectedEventJson = json_encode(array(
+            'class' => 'test.some.class',
+            'source' => 'some-source',
+            'props' => array(
+                'foo' => 'bar'
+            ),
+            'user' => 'some-user',
+            'timestamp' => '1531381642499'
+        ));
+        $expectedConnectTimeout = array('connect_timeout' => 2);
+
+        $stubHttpClient = $this->getMock('\Guzzle\Http\Client', array('post'));
+        $stubHttpClient->expects($this->once())->method('post')->with(
+            'http://example.com:3002/1/events',
+            $expectedHeaders,
+            $expectedEventJson,
+            $expectedConnectTimeout
+        )->will($this->returnValue($mockRequest));
+
+        $echoClient = $this->getMock('\echoclient\EchoClient', array('getPersonaClient', 'getHttpClient'));
+        $echoClient->expects($this->once())->method('getPersonaClient')->will($this->returnValue($stubPersonaClient));
+        $echoClient->expects($this->once())->method('getHttpClient')->will($this->returnValue($stubHttpClient));
+
+        $bSent = $echoClient->createEvent('some.class', 'some-source', array('foo'=>'bar'), 'some-user', '1531381642499');
+
+        $this->assertTrue($bSent);
+    }
+
+    function testCreateEvents()
+    {
+        $this->setRequiredDefines();
+
+        $stubPersonaClient = $this->getMock('\Talis\Persona\Client\Tokens', array(), array(), '', false);
+        $stubPersonaClient->expects($this->once())->method('obtainNewToken')->will($this->returnValue(array('access_token'=>'some-token')));
```

Comment:
Is there a minimum PHP version being supported?  Wondering if we can use the `[]` array literal syntax.